web_external/models/UserModel.js

```diff
@@ -33,39 +34,53 @@ UserModel.prototype.canAcceptTerms = function () {
     return this.get('permissions').acceptTerms === true;
 };
 
-UserModel.prototype.setAcceptTerms = function (successCallback) {
+UserModel.prototype.setAcceptTerms = function () {
+    const deferred = $.Deferred();
     restRequest({
         path: 'user/acceptTerms',
         type: 'POST'
-    }).done((resp) => {
+    })
+    .done((resp) => {
         if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
             // Directly update user permissions
             this.get('permissions').acceptTerms = true;
             this.trigger('change:permissions');
-            successCallback(resp);
+            deferred.resolve(resp);
+        } else {
+            // This should not fail
+            deferred.reject(resp);
         }
-        // This should not fail
+    })
+    .fail((resp) => {
+        deferred.reject(resp);
     });
+    return deferred.promise();
```

Comment:
I think we should be able to avoid this pattern of creating and resolving/rejecting a new Deferred. Something like:
````
    return restRequest({
        path: 'user/acceptTerms',
        type: 'POST'
    })
    .done((resp) => {
        if (_.has(resp, 'extra') && resp.extra === 'hasPermission') {
            // Directly update user permissions
            this.get('permissions').acceptTerms = true;
            this.trigger('change:permissions');
    })
````